packages/block-library/src/embed/index.js

```diff
@@ -55,18 +55,21 @@ export function getEmbedEdit( title, icon ) {
 			this.setAspectRatioClassNames = this.setAspectRatioClassNames.bind( this );
 			this.getResponsiveHelp = this.getResponsiveHelp.bind( this );
 			this.toggleResponsive = this.toggleResponsive.bind( this );
+			this.receivePreview = this.receivePreview.bind( this );
 
 			this.state = {
 				editingURL: false,
 				url: this.props.attributes.url,
 			};
 
-			this.maybeSwitchBlock();
+			if ( this.props.preview ) {
+				this.receivePreview();
+			}
 		}
 
-		componentWillUnmount() {
-			// can't abort the fetch promise, so let it know we will unmount
-			this.unmounting = true;
+		receivePreview() {
```

Comment:
`receivePreview` seems a bit of an odd name for what it arguably *rendering* or *changing attributes of* the preview. I don't know what `this.maybeSwitchBlock` does because it's another weird *maybe* function name, but would this be better-named as `this.updatePreview`? 🤷‍♂️

That or maybe just use `this.setAttributesFromPreview()` in place of calling `this.receivePreview()` and add `this. .maybeSwitchBlock()` to the end of `setAttributesFromPreview`. Hope that makes sense!

Reply:
It's code we need to run whenever we receive a preview, and to keep things small and the functions easy to read, `setAttributesFromPreview` and `maybeSwitchBlock` are split. `handleIncomingPreview` perhaps?

`maybe` means that calling that method might do what it says, but the conditions are complex and there's no guarantee. The method is short and documented. You can find the `maybe` prefix in the parser, in the popover component, in the async-generator middleware, in the block list, and in the format toolbar. And in WordPress core PHP code.

Reply:
I know it's elsewhere, but it breaks my brain. I think I want to launch a campaign to improve it. As @aduth has said elsewhere (I'm paraphrasing): "Just because we do it less optimally elsewhere doesn't mean we should here." 😏

But yes, it's not isolated to this patch. I like `handleIncomingPreview`, that makes more sense to me. 🤷‍♂️

Reply:
I was under the (possibly wrong) impression that it was an established pattern, as it's in both WP core and G, with some of the usage in G being added as recently as last week.

I'll change `receivePreview` to `handleIncomingPreview`, that does make it clearer :)